src/test-helper.js

```diff
@@ -577,16 +544,21 @@ function verifyDocumentDeleted(oldDoc, expectedAuthorization, expectedAccessAssi
 }
 
 function verifyDocumentRejected(doc, oldDoc, docType, expectedErrorMessages, expectedAuthorization) {
+  var syncFuncError = null;
   try {
     syncFunction(doc, oldDoc);
-    assert.fail('Document validation succeeded when it was expected to fail');
```

Comment:
If this statement executes, it throws an exception that is subsequently caught by the `catch` block and mishandled. The result is now verified outside of the try-catch to avoid the issue.